test/mocha/20-revocation.js

```diff
@@ -0,0 +1,367 @@
+const helpers = require('./helpers');
+const jsigs = require('jsonld-signatures');
+const {CapabilityDelegation} = require('ocapld');
+const {AsymmetricKey, CapabilityAgent, KmsClient, KeystoreAgent} =
+  require('webkms-client');
+const {Ed25519KeyPair} = require('crypto-ld');
+const {util: {uuid}} = bedrock;
+const {
+  purposes: {AssertionProofPurpose},
+  sign,
+  suites: {Ed25519Signature2018}
+} = jsigs;
+
+const KMS_MODULE = 'ssm-v1';
+
+describe('revocations API', () => {
+  let aliceCapabilityAgent;
+  let aliceKeystoreAgent;
+  let bobCapabilityAgent;
+  let bobKeystoreAgent;
+  let bobKey;
+  let carolCapabilityAgent;
+  let carolKey;
+  let carolKeystoreAgent;
+
+  before(async () => {
+    const secret = '40762a17-1696-428f-a2b2-ddf9fe9b4987';
+    const handle = 'testKey2';
+    aliceCapabilityAgent = await CapabilityAgent.fromSecret({secret, handle});
+
+    const keystore = await helpers.createKeystore(
+      {capabilityAgent: aliceCapabilityAgent});
+    const {httpsAgent} = brHttpsAgent;
+    const kmsClient = new KmsClient({httpsAgent});
+    aliceKeystoreAgent = new KeystoreAgent(
+      {capabilityAgent: aliceCapabilityAgent, keystore, kmsClient});
+  });
+
+  // generate a keystore for Bob
+  before(async () => {
+    const secret = '34f2afd1-34ef-4d46-a998-cdc5462dc0d2';
+    const handle = 'bobKey';
+    bobCapabilityAgent = await CapabilityAgent.fromSecret({secret, handle});
+    const keystore = await helpers.createKeystore(
+      {capabilityAgent: bobCapabilityAgent});
+    try {
+      const {httpsAgent} = brHttpsAgent;
+      const kmsClient = new KmsClient({httpsAgent});
+      bobKeystoreAgent = new KeystoreAgent(
+        {capabilityAgent: bobCapabilityAgent, keystore, kmsClient});
+    } catch(e) {
+      assertNoError(e);
+    }
+    try {
+      bobKey = await bobKeystoreAgent.generateKey(
+        {type: 'Ed25519VerificationKey2018', kmsModule: KMS_MODULE});
+    } catch(e) {
+      assertNoError(e);
+    }
+    await _setKeyId(bobKey);
+  });
+
+  // generate a keystore for Carol
+  before(async () => {
+    const secret = 'ae806cd9-2765-4232-b955-01e1024ac032';
+    const handle = 'carolKey';
+    const {httpsAgent} = brHttpsAgent;
+    // keystore in the kmsClient is set later
+    const kmsClient = new KmsClient({httpsAgent});
+    carolCapabilityAgent = await CapabilityAgent.fromSecret({
+      secret, handle, kmsClient
+    });
+    const keystore = await helpers.createKeystore(
+      {capabilityAgent: carolCapabilityAgent});
+    carolKeystoreAgent = new KeystoreAgent(
+      {capabilityAgent: carolCapabilityAgent, keystore, kmsClient});
+
+    carolKey = await carolKeystoreAgent.generateKey(
+      {type: 'Ed25519VerificationKey2018', kmsModule: KMS_MODULE});
+    await _setKeyId(carolKey);
+  });
+
+  // mock session authentication for delegations endpoint
+  let passportStub;
+  before(() => {
+    const actor = {
+      // FIXME: how does the account/actor ID, a urn:uuid: relate to
```

Comment:
@dlongley I think we discussed this, but can you remind me how this works?

Reply:
It doesn't. Actor is just the account that the user has authenticated with.